### PR TITLE
feat: Query struct with filter

### PR DESCRIPTION
**Pull request by JossDuff**
Adding filter fields to the query struct

### CODE REVIEW

File: hypersync-net-types/src/bloom_filter_wrapper.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+#[derive(Debug, Clone)]
+pub struct FilterWrapper(sbbf_rs_safe::Filter);
+
+impl PartialEq for FilterWrapper {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.as_bytes() == other.0.as_bytes()
```
Comment (JossDuff) [inline hypersync-net-types/src/bloom_filter_wrapper.rs:14]: Not sure if this is okay

Comment (ozgrakkurt) [reply hypersync-net-types/src/bloom_filter_wrapper.rs:14]: should be ok but we probably don't need it

File: hypersync-net-types/src/bloom_filter_wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+use hex::{decode, encode};
```
Comment (ozgrakkurt) [inline hypersync-net-types/src/bloom_filter_wrapper.rs:1]: should use faster-hex and maybe even prefixed hex encoding like the functions in `format` crate

**Review by ozgrakkurt — COMMENTED** [review]
looks good except hex serialisation

**Comment by ozgrakkurt** [comment]
should be minor version bump on client as well. so 0.10 -> 0.11

File: hypersync-net-types/src/bloom_filter_wrapper.rs (hunk 2 above)
Comment (JossDuff) [reply hypersync-net-types/src/bloom_filter_wrapper.rs]: Moved it into format and using the encoding/decoding from `Data`

File: hypersync-net-types/src/bloom_filter_wrapper.rs (hunk 1 above)
Comment (JossDuff) [reply hypersync-net-types/src/bloom_filter_wrapper.rs]: Just had to include it because removing `PartialEq` from Query breaks a bunch of things

**Review by ozgrakkurt — APPROVED** [review]
💪🚀

**Pull request merged**